include/range/v3/view/group_by.hpp

```diff
@@ -53,56 +54,52 @@ namespace ranges
     private:
         friend range_access;
         Rng rng_;
+        // cached version of the end of the first subrange / start of the second subrange
```

Comment:
This comment is wrong. `second_` may be any of those, given the right inputs, but not necessarily.

Reply:
I do not think the comment is wrong.

Reply:
In the other cases, it is the second element of the first subrange, isn't it?

Reply:
No. It's always the sentinel of the first subrange. Which would be the beginning of the second subrange, if there is one.

Reply:
How can that be? I'm seeing that the logic in `next()` and `begin_cursor()` is the same.

Reply:
Consider the range `[1, 2, 2, 3, 3, 4, 5]` and the predicate `==`. `begin_cursor()`
will look like:

```
[1, 2, 2, 3, 3, 4, 5]
 ↑  ↑               ↑
 |  next_cur_       last_
 cur_
```

and on the next iteration will look like:

```
[1, 2, 2, 3, 3, 4, 5]
    ↑     ↑         ↑
    |   next_cur_   last_
   cur_                 
```

The `view`'s `second_` member is the initial value of the `cursor`'s `next_cur_` member.

Reply:
Yeah, the logic is the same - we're advancing by a whole subrange (not a single element in it).